central/policy/service/validator_test.go

```diff
@@ -1021,3 +991,114 @@ func (s *PolicyValidatorTestSuite) TestValidateEnforcement() {
 		})
 	}
 }
+
+func (s *PolicyValidatorTestSuite) TestValidateNodeEventSource() {
+	testCases := []struct {
+		description string
+		p           *storage.Policy
+		errExpected bool
+	}{
+		{
+			description: "Node policy with no fields",
+			p:           booleanPolicyWithFields(storage.LifecycleStage_RUNTIME, storage.EventSource_NODE_EVENT, nil),
+			errExpected: true,
+		},
+		{
+			description: "Node policy with deploy-time fields only",
+			p: booleanPolicyWithFields(storage.LifecycleStage_RUNTIME, storage.EventSource_NODE_EVENT,
+				map[string]string{
+					fieldnames.ImageTag:   "latest",
+					fieldnames.VolumeName: "BLAH",
+				}),
+			errExpected: true,
+		},
+		{
+			description: "Node policy with process field (should be invalid)",
+			p: booleanPolicyWithFields(storage.LifecycleStage_RUNTIME, storage.EventSource_NODE_EVENT,
+				map[string]string{
+					fieldnames.ProcessName: "malicious-process",
+				}),
+			errExpected: true,
+		},
+		{
+			description: "Node policy with wrong lifecycle stage (build)",
+			p:           booleanPolicyWithFields(storage.LifecycleStage_BUILD, storage.EventSource_NODE_EVENT, nil),
+			errExpected: true,
+		},
+		{
+			description: "Node policy with wrong lifecycle stage (deploy)",
+			p:           booleanPolicyWithFields(storage.LifecycleStage_DEPLOY, storage.EventSource_NODE_EVENT, nil),
+			errExpected: true,
+		},
+		{
+			description: "Node policy with valid NodeFilePath field",
+			p: booleanPolicyWithFields(storage.LifecycleStage_RUNTIME, storage.EventSource_NODE_EVENT,
+				map[string]string{
+					fieldnames.NodeFilePath: "/etc/passwd",
+				}),
+		},
```

Comment:
nit: this looks like the only "happy path" test in this suite, and it's buried in the middle. I prefer to group all happy-path tests at the beginning of a suite.